ci/docker/arch-linux

```diff
@@ -0,0 +1,17 @@
+FROM public.ecr.aws/docker/library/archlinux:latest 
+
+RUN pacman -Sy --noconfirm git 
+RUN git clone --recurse-submodules https://github.com/aws/aws-sdk-cpp.git
+RUN pacman -Sy --noconfirm \
+    cmake \
+    ninja \
+    clang \
+    curl \
+    zip
+
+RUN CC=/usr/bin/clang CXX=/usr/bin/clang++ cmake -Saws-sdk-cpp -Baws-sdk-cpp/build -GNinja \
+    -DBUILD_ONLY=lambda \
+    -DUSE_OPENSSL=OFF \ 
```

Comment:
This is a fun one - basically, because reasons (https://github.com/aws/aws-sdk-cpp/issues/1910), including the current version of AWS SDK into the project (as the tests do) - CMake barfs at test time (`./ci/codebuild/build.sh` in `arch-linux.yml`) complaining about not being able to find openssl, enough though it was found just fine when doing the SDK build + install! For the other build environments we have, something like `yum install -y openssl-static` is resolving the dependency, but an equivalent package is not available (that I could find anyways) in this distro.

So... the workaround I've been doing is using the experimental S2N support in the SDK, triggered by this flag. This works fine, but also brought along the requirement to use `clang` instead of `gcc` due to differences in warnings/errors reported on when Came compiled with this distro's gcc/g++.

🤷‍♂️

Reply:
Yikes!
You should probably put that blurb of reasoning as a comment in the docker file.
I wish the C++ SDK never adopted the CRT. Oh Well!

Reply:
Added a shorter version of this comment

Reply:
In Arch linux, the AUR has a package [openssl-static](https://aur.archlinux.org/packages/openssl-static).

Reply:
If desired, it could be installed with:
```
git clone --depth 1 https://aur.archlinux.org/openssl-static.git
cd openssl-static
makepkg -si --noconfirm --skippgpcheck
```
However, it will build openssl from source, which takes quite a while.